File: client/src/types/bot.ts
export type BotStatus = 'IDLE' | 'PROCESSING';

export interface Bot {
  id: number;
  status: BotStatus;
  currentOrderId?: number;
  processingStartTime?: string;
}

export interface CreateBotResponse {
  bot: Bot;
}

export interface RemoveBotResponse {
  bot: Bot | null;
}

File: internal/bot.go
package internal

import "context"

type BotStatus string

const (
	Idle BotStatus = "IDLE"
	Busy BotStatus = "PROCESSING"
)

type Bot struct {
	ID           int
	status       BotStatus
	currentOrder *Order
	cancel       context.CancelFunc
}

// Status returns the current bot status.
func (b *Bot) Status() BotStatus { return b.status }
